### PR TITLE
Adding ImmersiveReaderView for UWP

### PR DESCRIPTION
**Pull request by dpaulino**
Hi all,

I've recently become a big fan of Immersive Reader and I wanted to integrate it into my UWP applications. So, I created this new custom UWP control to help with it. I welcome all feedback to this pull request, and I hope that we can work together to eventually add this UWP control into the SDK. 

Additionally, if this PR is completed, I'd be happy to work with someone on setting up the pipeline for publishing this to Nuget. I have already tested this process out by installing the ImmersiveReaderView control via local nupkg files. It works great. 

Thanks,
Daniel

### CODE REVIEW

File: uwp/ImmersiveReader/script.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿<html>
```
Comment (metanMSFT) [inline uwp/ImmersiveReader/script.txt]: Can this be renamed to script.html?

File: uwp/ImmersiveReader/script.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+<head>
+    <meta charset='utf-8'>
+    <title>Immersive Reader Example: Document</title>
```
Comment (metanMSFT) [inline uwp/ImmersiveReader/script.txt]: Can remove the `title`

File: uwp/ImmersiveReader/script.txt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    <script type='text/javascript' src='https://contentstorage.onenote.office.net/onenoteltir/immersivereadersdk/immersive-reader-sdk.0.0.3.js'></script>
```
Comment (metanMSFT) [inline uwp/ImmersiveReader/script.txt]: With the release of v1.0.0, we can use `immersive-reader-sdk.1.0.0.js` instead.

File: uwp/ImmersiveReader/ImmersiveReaderView.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+            var token = await GetTokenAsync();
+            if (string.IsNullOrWhiteSpace(token))
+            {
+                return;
+            }
+
+            text = text.Replace("|TOKEN|", token);
+            text = text.Replace("|YOUR_SUB_DOMAIN|", this.Subdomain);
+            text = text.Replace("|CONTENT|", this.ReaderContent);
```
Comment (metanMSFT) [inline uwp/ImmersiveReader/ImmersiveReaderView.xaml.cs]: Single-quotes in the content need to be escaped

File: uwp/ImmersiveReader/script.txt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                hideExitButton: true,
+                allowFullscreen: false
+            }
+
+            ImmersiveReader.launchAsync('|TOKEN|', '|YOUR_SUB_DOMAIN|', data, options)
+                .then(() => {
+                    console.log('success');
+                }, (error) => {
+                    console.log('error! ' + error.message + ' ' + error.name);
+                });
```
Comment (metanMSFT) [inline uwp/ImmersiveReader/script.txt]: Can be removed.

File: uwp/ImmersiveReader/script.txt [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    <meta name='viewport' content='width=device-width, initial-scale=1'>
+</head>
+
+<body onload="handleLaunchImmersiveReader()">
+
+    <script type='text/javascript'>
+
+        async function handleLaunchImmersiveReader() {
+            const data = {
+                title: ' ',
```
Comment (metanMSFT) [inline uwp/ImmersiveReader/script.txt]: Should support title as well, e.g. `StartAsync(string title, string content)`

File: uwp/README.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+Usage of this SDK requires an Azure subscription to Immersive Reader. Follow [these instructions](https://docs.microsoft.com/azure/cognitive-services/immersive-reader/how-to-create-immersive-reader) to create an Immersive Reader resource and configure Azure Active Directory authentication. Save the output of your session into a text file for future reference.
+
+For sample usage, launch the `ImmersiveReader.sln` file in Visual Studio 2019. Ensure that you have the UWP development tooling installed. Note that ImmersiveReaderView only supports the Fall Creator's Update (16299) or higher.
+
+Press `F5` to run the sample application. Enter your azure information into the sample app and click the `Start immersive reader` button.
+
+## Nuget
+
+TODO: use the nuspec file to publish this component in Nuget to optimize the developer workflow.
```
Comment (metanMSFT) [inline uwp/README.md]: Can remove

File: uwp/README.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+The ImmersiveReaderView control is a Windows Runtime component that allows you to easily and quickly integrate the [Immersive Reader](https://azure.microsoft.com/services/cognitive-services/immersive-reader/) into your UWP application.
+
+## Usage
+
+Usage of this SDK requires an Azure subscription to Immersive Reader. Follow [these instructions](https://docs.microsoft.com/azure/cognitive-services/immersive-reader/how-to-create-immersive-reader) to create an Immersive Reader resource and configure Azure Active Directory authentication. Save the output of your session into a text file for future reference.
+
+For sample usage, launch the `ImmersiveReader.sln` file in Visual Studio 2019. Ensure that you have the UWP development tooling installed. Note that ImmersiveReaderView only supports the Fall Creator's Update (16299) or higher.
+
+Press `F5` to run the sample application. Enter your azure information into the sample app and click the `Start immersive reader` button.
```
Comment (metanMSFT) [inline uwp/README.md]: Nit: capitalization of Azure

File: uwp/ImmersiveReader/ImmersiveReaderView.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+            set => SetValue(ClientSecretProperty, value);
+        }
+
+        private static DependencyProperty ClientSecretProperty = DependencyProperty.Register(
+            "ClientSecret",
+            typeof(string),
+            typeof(ImmersiveReaderView),
+            null);
+
+        public async void Start()
```
[the remark this answers is not in the document]
Comment (dpaulino) [reply uwp/ImmersiveReader/ImmersiveReaderView.xaml.cs]: @metanMSFT there seems to be a restriction with Windows Runtime components that restricts the use of `Task` as a return type of public methods. 
image

[This msdn post](https://social.msdn.microsoft.com/forums/windowsapps/en-US/5370453f-9192-4593-9105-11cd69c48894/how-to-use-task-with-background-task) recommends returning `IAsyncAction` instead. 

I'll go ahead with using `IAsyncAction` unless you have other thoughts.

[the remark this answers is not in the document]
Comment (dpaulino) [reply uwp/ImmersiveReader/ImmersiveReaderView.xaml.cs]: Using IAsyncAction seems to be working. Resolving comment.

**Comment by dpaulino** [comment]
@metanMSFT thanks for your review! I feel I've addressed your comments now. Let me know what you think

**Comment by metanMSFT** [comment]
@dpaulino Looks good, my only comment now is to move the code into the js/samples folder

**Comment by dpaulino** [comment]
I've moved it to the samples directory now. Let me know what you think

**Comment by metanMSFT** [comment]
@dpaulino Merged! Thank you for your contribution!

**Pull request merged**